src/core/net/srp_client.hpp

```diff
@@ -96,7 +96,7 @@ class Client : public InstanceLocator, private NonCopyable
      * This type represents a TXT record entry representing a key/value pair (RFC 6763 - section 6.3).
      *
      */
-    class TxtEntry : public otSrpTxtEntry
+    class TxtEntry : public otDnsTxtEntry
```

Comment:
Since the `oDnsTxtEntry` is now a common public DNS related type, shall we also move the `TxtEntry` definition from under `Srp::Clinet` here into a common internal header, say `dns_header.hpp` (or a new common header for use by core). Since we have this c++ type in the core, I think would better to use in SRP server as well (instead of directly using the public C `otDnsTxtEntry` definition)

Another idea/suggestion is to see if we can somehow define the method that does the iteration over TXT entries (and the definition of iterator) into the `TxtEntry` class as well.

For example, the method can be like:
```
otError TxtEntry::GetNextEntry(uint8_t *aTxtData, uint16_t aTxtLength, Iterator &aIterator, TxEntry &aEntry);
```
i.e., we pass in the encoded TXT data and its length and the current iterator value and it then populates the next entry.

Reply:
Thanks, I forgot about the `TxtEntry` class in `Srp::Client`. You proposal sounds good to me. I will move the iterator method to the `TxtEntry` class and move the `TxtEntry` into `dns_headers.hpp`.

Reply:
Actually, I added the `GetNextEntry` function to `TxtRecord` as it looks like a more appropriate place :)